controllers/authController.js

```diff
@@ -0,0 +1,51 @@
+const User = require("../models/userModel");
+const catchAsync = require("../utils/catchAsync");
+const jwt = require("jsonwebtoken");
+const AppError = require("../utils/appError");
+
+const signToken = (id) =>
+  jwt.sign(
+    {
+      id,
+    },
+    process.env.JWT_SECRET,
+    {
+      expiresIn: process.env.JWT_EXPIRES_IN,
+    }
+  );
+exports.signup = catchAsync(async (req, res, next) => {
+  const newUser = await User.create({
+    name: req.body.name,
+    email: req.body.email,
+    password: req.body.password,
+    passwordConfirm: req.body.passwordConfirm,
+  });
+  const token = signToken(newUser._id);
+  res.status(201).json({
+    status: "success",
+    token,
+    data: {
+      user: newUser,
+    },
+  });
+});
+
+exports.login = catchAsync(async (req, res, next) => {
+  const { email, password } = req.body;
+  // 1. check if email and password is not empty
+  if (!email || !password) {
```

Comment:
Also, there's no way you are ensuring the email is even a valid email or the password meets any specific constraint.

This is not good enough. Read about this [library](https://express-validator.github.io/docs/category/guides) and implement to ensure email received has a valid  email structure